### PR TITLE
Fix Style/HashAsLastArrayItem with no_braces for empty hash

### PR DESCRIPTION
**Pull request by fsateler**
When `EnforcedStyle` is set to `no_braces`, rubocop was attempting to
correct `[1, {}]` into `[1, ]`, which is not valid. An empty hash cannot
omit the braces.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x]  Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
The change looks good, but for some reason the CI is failing.

**Comment by fsateler** [comment]
I don't understand how the failures can be related to the PR...

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 * [#8664](https://github.com/rubocop-hq/rubocop/issues/8664): Fix a false positive for `Naming/BinaryOperatorParameterName` when naming multibyte character method name. ([@koic][])
 * [#8604](https://github.com/rubocop-hq/rubocop/issues/8604): Fix a false positive for `Bundler/DuplicatedGem` when gem is duplciated in condition. ([@tejasbubane][])
 * [#8671](https://github.com/rubocop-hq/rubocop/issues/8671): Fix an error for `Style/ExplicitBlockArgument` when using safe navigation method call. ([@koic][])
+* [#8681](https://github.com/rubocop-hq/rubocop/pull/8681): Fix an error for `Style/HashAsLastArrayItem` with `no_braces` for empty hash.  ([@fsateler][])
```
Comment (koic) [inline CHANGELOG.md:25]: Can you remove the following redundant blank character?

```suggestion
* [#8681](https://github.com/rubocop-hq/rubocop/pull/8681): Fix an error for `Style/HashAsLastArrayItem` with `no_braces` for empty hash. ([@fsateler][])
```

**Comment by koic** [comment]
> I don't understand how the failures can be related to the PR...

The following spec is failing.

```console
Failures:

  1) RuboCop Project changelog has link definitions for all implicit links
     Failure/Error:
       expect(changelog.include?("[#{name}]: http"))
         .to be(true), "CHANGELOG.md is missing a link for #{name}. " \
                       'Please add this link to the bottom of the file.'

       CHANGELOG.md is missing a link for @fsateler. Please add this link to the bottom of the file.
     # ./spec/project_spec.rb:125:in `block (4 levels) in <top (required)>'
     # ./spec/project_spec.rb:124:in `each'
     # ./spec/project_spec.rb:124:in `block (3 levels) in <top (required)>'
     # tasks/spec_runner.rake:32:in `block in run_specs'
     # tasks/spec_runner.rake:44:in `with_encoding'
     # tasks/spec_runner.rake:28:in `run_specs'
     # tasks/spec_runner.rake:116:in `block in <top (required)>'

Finished in 2 minutes 29.1 seconds (files took 3.26 seconds to load)
13875 examples, 1 failure, 10 pending
```

https://github.com/rubocop-hq/rubocop/pull/8681/checks?check_run_id=1111352141#step:6:158

Can you add your account to the bottom of CHANGELOG.md?

```diff
diff --git a/CHANGELOG.md b/CHANGELOG.md
index 5936a4678..01b0bb8d8 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4856,3 +4856,4 @@
 [@jaimerave]: https://github.com/jaimerave
 [@Skipants]: https://github.com/Skipants
 [@sascha-wolf]: https://github.com/sascha-wolf
+[@fsateler]: https://github.com/fsateler
```

https://github.com/rubocop-hq/rubocop/blob/v0.90.0/CONTRIBUTING.md#changelog-entry-format

**Pull request merged**